local.tf

```diff
@@ -0,0 +1,566 @@
+terraform {
+  required_version = "~> 0.12.25"
+  backend "local" {}
+}
+
+provider "aws" {
+  region = "us-west-2"
+}
+
+##########################################
+# Variables
+##########################################
+
+variable "global_environment_name" {
+  description = "A globally unique environment name for S3 buckets."
+  type        = string
+}
+
+##########################################
+# Data
+##########################################
+
+data "aws_region" "current" {
+}
+
+data "aws_availability_zones" "available" {
+}
+
+##########################################
+# VPC resources
+##########################################
+
+resource "aws_vpc" "wandb" {
+  cidr_block           = "10.10.0.0/16"
+  enable_dns_support   = true
+  enable_dns_hostnames = true
+
+  tags = {
+    "Name" = "wandb"
+    "kubernetes.io/cluster/wandb" = "shared"
+  }
+}
+
+resource "aws_subnet" "wandb" {
+  count = 2
+
+  availability_zone       = data.aws_availability_zones.available.names[count.index]
+  cidr_block              = "10.10.${count.index}.0/24"
+  vpc_id                  = aws_vpc.wandb.id
+  map_public_ip_on_launch = true
+
+  tags = {
+    "Name" = "wandb-${count.index}"
+    "kubernetes.io/cluster/wandb" = "shared"
+  }
+}
+
+resource "aws_internet_gateway" "wandb" {
+  vpc_id = aws_vpc.wandb.id
+
+  tags = {
+    Name = "wandb-gateway"
+  }
+}
+
+resource "aws_route_table" "wandb" {
+  vpc_id = aws_vpc.wandb.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.wandb.id
+  }
+}
+
+resource "aws_route_table_association" "wandb_workers" {
+  count = 2
+
+  subnet_id      = aws_subnet.wandb[count.index].id
+  route_table_id = aws_route_table.wandb.id
+}
+
+##########################################
+# EKS resources
+##########################################
+
+resource "aws_security_group" "eks_master" {
+  name        = "wandb-eks-master"
+  description = "Cluster communication with worker nodes"
+  vpc_id      = aws_vpc.wandb.id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "wandb-eks-master"
+  }
+}
+
+resource "aws_eks_cluster" "wandb" {
+  name     = "wandb"
+  role_arn = aws_iam_role.wandb_cluster_role.arn
+  version  = "1.15"
+
+  vpc_config {
+    security_group_ids = [aws_security_group.eks_master.id]
+    subnet_ids         = aws_subnet.wandb.*.id
+  }
```

Comment:
What's the best practice for connecting this cluster to an existing VPC?

Reply:
This might be a bit tricky to set up, but I can probably create a variable that allows you to specify your own VPC ARN and then only conditionally create a new VPC. Alternatively, customers can set up VPC peering.